pkg/scheduler/framework/runtime/framework.go

```diff
@@ -904,29 +904,29 @@ func (f *frameworkImpl) runPreScorePlugin(ctx context.Context, pl framework.PreS
 // stores for each scoring plugin name the corresponding NodeScoreList(s).
 // It also returns *Status, which is set to non-success if any of the plugins returns
 // a non-success status.
-func (f *frameworkImpl) RunScorePlugins(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodes []*v1.Node) (ps framework.PluginToNodeScores, status *framework.Status) {
+func (f *frameworkImpl) RunScorePlugins(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodes []*v1.Node) (ns []framework.NodePluginScores, status *framework.Status) {
 	startTime := time.Now()
 	defer func() {
 		metrics.FrameworkExtensionPointDuration.WithLabelValues(score, status.Code().String(), f.profileName).Observe(metrics.SinceInSeconds(startTime))
 	}()
-	pluginToNodeScores := make(framework.PluginToNodeScores, len(f.scorePlugins))
-	for _, pl := range f.scorePlugins {
-		pluginToNodeScores[pl.Name()] = make(framework.NodeScoreList, len(nodes))
-	}
+	allNodePluginScores := make([]framework.NodePluginScores, len(nodes))
+	pluginToNodeScores := make([]framework.NodeScoreList, len(f.scorePlugins))
 	ctx, cancel := context.WithCancel(ctx)
 	errCh := parallelize.NewErrorChannel()
 
-	// Run Score method for each node in parallel.
-	f.Parallelizer().Until(ctx, len(nodes), func(index int) {
-		for _, pl := range f.scorePlugins {
-			nodeName := nodes[index].Name
+	// Run Score method of each score plugin in parallel.
+	f.Parallelizer().Until(ctx, len(f.scorePlugins), func(i int) {
+		pl := f.scorePlugins[i]
+		pluginToNodeScores[i] = make(framework.NodeScoreList, len(nodes))
+		for index, node := range nodes {
+			nodeName := node.Name
```

Comment:
By changing the return type of RunScorePlugins, we don't need to use a map for `pluginToNodeScores` and can use a slice instead. And we can remove initializing pluginToNodeScores in `for` loop by changing `f.Parallelizer().Until` to run for every score plugin, not for every node. 

This seems to contribute to improved performance? It's a little hard to say because the changes are so small, but it does seem to improve performance:

```
#108534
metric: scheduler_pod_scheduling_duration_seconds, avarage of avarages: 974.6977533840003
metric: scheduler_framework_extension_point_duration_seconds Score, avarage of avarages: 1.4469456398
metric: scheduler_framework_extension_point_duration_seconds Filter, avarage of avarages: 0.4670207798
metric: scheduler_e2e_scheduling_duration_seconds, avarage of avarages: 4.251927499100001
metric: SchedulingThroughput, avarage of avarages: 459.0166666666666

===

# master
metric: scheduler_pod_scheduling_duration_seconds, avarage of avarages: 1052.5383911987003
metric: scheduler_framework_extension_point_duration_seconds Score, avarage of avarages: 1.2655535807999998
metric: scheduler_framework_extension_point_duration_seconds Filter, avarage of avarages: 0.5263045056999999
metric: scheduler_e2e_scheduling_duration_seconds, avarage of avarages: 4.791027555099997
metric: SchedulingThroughput, avarage of avarages: 453.4666666666666
```

(The way of running scheduler_perf is the same as https://github.com/kubernetes/kubernetes/pull/108494#issuecomment-1087386857)